### PR TITLE
PR 61

### PR DESCRIPTION
**Pull request by ericsorenson**
### Stories & Bugs
- [7650 BUG: Completed Messages Still Displaying in **SECTION** inbox and as "In Progress" within a case](https://github.com/flexion/ef-cms/issues/7650) 
- [7639 BUG: White Screen when clicking on message](https://github.com/flexion/ef-cms/issues/7639) 
- [7608 BUG: Docket QC Complete and Send Message - Not Routing to Paper Service Screen or Generating Paper Service PDF ](https://github.com/flexion/ef-cms/issues/7608) 
- [7584 BUG: Electronic signatures not viewable in PDF preview](https://github.com/flexion/ef-cms/issues/7584) 
- [7580 BUG: Public docket record: Not displaying parenthetical information on docket entries  ](https://github.com/flexion/ef-cms/issues/7580) 
- [7579 BUG: Public docket record:  Displaying wrong event code for Lodged documents](https://github.com/flexion/ef-cms/issues/7579) 
- [7494 BUG: Complete QC and Send Message: Updates to document title are not appearing in send message modal ](https://github.com/flexion/ef-cms/issues/7494) 
- [7493 BUG: Create/Send Message modal : Attach document dropdown does not show "Additional Info" as part of the document title   ](https://github.com/flexion/ef-cms/issues/7493) 
- [7492 BUG: Updating field in concatenated document title during QC or Edit Docket Entryis not updating document title ](https://github.com/flexion/ef-cms/issues/7492) 
- [7445 Admissions Clerk: Edit Practitioner profile when no email](https://github.com/flexion/ef-cms/issues/7445) 

### Refactoring
- Increase reliability of integration tests
- Review AWS IAM Permissions

### Data Migrations
- None

### Manual Deployment Steps
- None

### CODE REVIEW

**Comment by klohman** [comment]
## PR 61 Release Notes 

### Internal 
**Improvement**: Admissions clerks can edit the profile for a practitioner without an email address.
* Clerks can make changes to practitioner information for practitioners with or without an email address on record. 

**Bug Fix**: Docket clerks will not be prompted to print paper service after completing QC and sending a message. 
* When completing QC and sending a message for a document to be served via paper service, clerks are now prompted immediately to print and mail the necessary documents. 
 
**Bug Fix**: Document titles will now display correctly across the application 
* Document titles were missing any “additional info” text when attaching a document to a message. The full document title now displays in the dropdown to attach a document. 
* Updates to the document title during the QC process were not being reflected when sending a message or being updated accurately on the Docket Record. Any updates made to a document title during the QC process reflect immediately on messages and on the Docket Record. 

**Bug Fix**: Electronic signatures are now visible in all PDF previews
*  Some PDF previews were not properly displaying electronic signatures on documents. Documents uploaded to the system with electronic signatures now display the signature in all PDF previews 
* Parties filing documents on mobile devices do not have access to PDF previews prior to submitting their documents. 

**Bug Fix**: Errors when clicking on messages has been resolved 
* Some users were encountering a white screen when clicking on their messages, which required to refresh to view the message. Users are now able to access their messages without refresh directly from their inboxes.   

**Bug Fix**: Completed messages will no longer display in any inbox or In Progress queue 
* Completed messages were still displaying in the section Inbox and in the In Progress tab on the case. 
* When a message is completed, it is no longer display in any individual or section Inboxes, or in the In Progress tab under Case Messages. Completed messages will display in the Completed tabs. 

**Bug Fix**: Public docket record now displays accurate event codes and filings and proceedings 
* The public view of the Docket Record was displaying an incorrect event code for lodged documents. All lodged documents now show on the Docket Record with the correct event code of MISCL.
* The public view of the Docket Record was not displaying the complete entry text under Filings and Proceedings. Parenthetical text (i.e. Attachments, Certificate of Service, Objections) now display as part of the Filings and Proceedings for all views of the Docket Record. 

### External / Public  
**Bug Fix**: Electronic signatures are now visible in all PDF previews
*  Some PDF previews were not properly displaying electronic signatures on documents. Documents uploaded to the system with electronic signatures now display the signature in all PDF previews 
* Parties filing documents on mobile devices do not have access to PDF previews prior to submitting their documents. 

**Bug Fix**: Public docket record now displays accurate event codes and filings and proceedings 
* The public view of the Docket Record was displaying an incorrect event code for lodged documents. All lodged documents now show on the Docket Record with the correct event code of MISCL.
* The public view of the Docket Record was not displaying the complete entry text under Filings and Proceedings. Parenthetical text (i.e. Attachments, Certificate of Service, Objections) now display as part of the Filings and Proceedings for all views of the Docket Record.

File: iam/terraform/environment-specific/main/lambda.tf [hunk 1]
```diff
@@ -119,10 +170,10 @@ resource "aws_iam_role_policy" "lambda_policy" {
...
             "Effect": "Allow",
             "Action": [
-                "sqs:SendMessage",
-                "sqs:ReceiveMessage",
                 "sqs:DeleteMessage",
-                "sqs:GetQueueAttributes"
+                "sqs:GetQueueAttributes",
+                "sqs:ReceiveMessage",
+                "sqs:SendMessage"
             ],
```
Comment (mmarcotte) [inline iam/terraform/environment-specific/main/lambda.tf:177]: Thanks for all of the security tightening above.

@lauraGgit will appreciate it.

File: web-client/integration-tests/docketClerkExternalDocumentQCWorkflow.test.js [hunk 2]
```diff
@@ -241,5 +279,181 @@ describe('Create a work item', () => {
...
+    expect(test.getState('currentPage')).toEqual('EditDocketEntry');
+
+    await test.runSequence('updateDocketEntryFormValueSequence', {
+      key: 'freeText',
+      value: 'break the test again',
+    });
+
+    await test.runSequence('completeDocketEntryQCSequence');
+
+    expect(test.getState('validationErrors')).toEqual({});
```
Comment (mmarcotte) [inline web-client/integration-tests/docketClerkExternalDocumentQCWorkflow.test.js:431]: What are your thoughts about making some of these integration tests actually send bad information along the way and generate some validation errors? Seems like that might be something helpful to test before trying again.

File: web-client/src/presenter/actions/updateMessageModalAfterQCAction.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  const generatedDocumentTitle = applicationContext
+    .getUtilities()
+    .getDocumentTitleWithAdditionalInfo({ docketEntry });
+
+  store.set(state.modal.form.subject, generatedDocumentTitle);
+
+  store.set(state.modal.form.attachments, [
+    {
+      documentId,
```
Comment (mmarcotte) [inline web-client/src/presenter/actions/updateMessageModalAfterQCAction.js]: Should we be introducing new `documentId` references in the code base? I thought we had made the switch to using `docketEntryId` for these items.

**Comment by mmarcotte** [comment]
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [ ] If changes were made in the UI (if not, skip):
  - [x] If desired, run `docker system prune` to remove any unused docker images from previous code review.
  - [x] Fetch the pull request for the sprint (e.g., `git fetch origin pull/{PR #}/head:sprint-{Sprint #}`), and then switch to that branch (e.g. `git checkout sprint-{Sprint #}`)
  - [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
	- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
	- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
	- [x] test in Mobile Safari and Mobile Chrome (or an emulator like Chrome DevTools), with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
	- [x] user-visible changes (including API users like the IRS) are documented in CHANGELOG.md (which is linked from US Tax Court website).
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: web-client/src/presenter/actions/updateMessageModalAfterQCAction.js (hunk 3 above)
Comment (ericsorenson) [reply web-client/src/presenter/actions/updateMessageModalAfterQCAction.js]: Good point, `documentId` is still in the codebase more than I'd like. I've created a backlog item. https://trello.com/c/GIy6A81K

File: web-client/integration-tests/docketClerkExternalDocumentQCWorkflow.test.js (hunk 2 above)
Comment (ericsorenson) [reply web-client/integration-tests/docketClerkExternalDocumentQCWorkflow.test.js:431]: Updated!

**Review by mmarcotte — APPROVED** [review]

**Pull request merged**